### PR TITLE
Speed up notifications bell (and other improvements)

### PR DESCRIPTION
**Pull request by thevoiceofzeke**
[MUMUP-3008](https://jira.doit.wisc.edu/jira/browse/MUMUP-3008): "As a user I would like notification indicators to reflect reality quickly so that I do not get the wrong impression from their state prior to their data loading."

**In this PR**:
- Notifications bell now shows a loading spinner while retrieving notifications
- No more UI delay when dismissing notifications/announcements from menus
- Empty state for notifications menu
- Notifications menu, mascot menu, and announcement popup now properly implement the `actionButton` and `moreInfoButton`
- Priority notifications in menu now have some spacing between icon and message

#### Notifications loading
notifications-loading-demo

#### Dismissing notifications
notifications-demo

### Screenshots

screen shot 2017-08-23 at 11 14 43 am
screen shot 2017-08-23 at 11 14 58 am
screen shot 2017-08-23 at 11 38 49 am

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

File: uw-frame-components/portal/messages/controllers.js [hunk 1]
```diff
@@ -210,6 +211,8 @@ define(['angular'], function(angular) {
             + (vm.notifications.length === 0
               ? 'no' : vm.notifications.length)
             + ' notifications';
+
+          vm.isLoading = false;
```
Comment (ChristianMurphy) [inline uw-frame-components/portal/messages/controllers.js]: :question: If loading fails will the loading spinner keep going forever?

**Comment by Doug-Reed** [comment]
:+1:

File: uw-frame-components/portal/messages/controllers.js (hunk 1 above)
Comment (vertein) [inline uw-frame-components/portal/messages/controllers.js]: Could add it to the failure as well

File: uw-frame-components/portal/messages/partials/announcement-popup-template.html [hunk 2]
```diff
@@ -11,8 +11,8 @@
   <md-dialog-content>
     <div class="md-dialog-content features-popup" layout="column" layout-align="center center">
       <img ng-src="{{ latestAnnouncement.featureImageUrl }}" aria-describedby="featureContent" hide-xs>
-      <p id="featureContent" class="sr-only" hide-xs>{{ latestAnnouncement.description }}</p>
-      <p hide-gt-xs>{{ latestAnnouncement.description }}</p>
+      <p id="featureContent" class="sr-only">{{ latestAnnouncement.description }}</p>
+      <p>{{ latestAnnouncement.description }}</p>
```
Comment (apetro) [inline uw-frame-components/portal/messages/partials/announcement-popup-template.html]: Could this `<p>` pick up the `id="featureContent" and the previous line's `<p` currently with that id simply be deleted?

File: uw-frame-components/portal/messages/partials/notifications-bell.html [hunk 3]
```diff
@@ -18,73 +18,81 @@
...
+        <md-menu-item ng-repeat="notification in vm.notifications | orderBy:'priority' | limitTo:3"
+                      class="top-bar-menu-item">
+          <!-- NOTIFICATION CONTENT -->
+          <div class="menu-item-content" layout="row" layout-align="space-between center">
+            <div flex="80" layout="column" layout-align="center start">
+              <!-- REGULAR MESSAGE -->
+              <span ng-if="!notification.priority">{{ notification.title }}</span>
+              <!-- PRIORITY MESSAGE -->
+              <div layout="row" layout-align="start center" ng-if="notification.priority">
+                <md-icon flex="10">priority_high</md-icon>
```
Comment (apetro) [inline uw-frame-components/portal/messages/partials/notifications-bell.html]: This icon isn't just decorative; it's conveying meaning about the importance of the notification it adorns. Needs some kind of alt text?

**Comment by davidmsibley** [comment]
I wouldn't let this stop you: https://www.traviscistatus.com/incidents/3q9c97727678

**Pull request merged**